apis/v1beta1/httproute_types.go

```diff
@@ -263,6 +263,56 @@ type HTTPRouteRule struct {
 	// +optional
 	// +kubebuilder:validation:MaxItems=16
 	BackendRefs []HTTPBackendRef `json:"backendRefs,omitempty"`
+
+	// Timeouts defines the timeouts that can be configured for an HTTP request.
+	//
+	// Support: Extended
+	//
+	// +optional
+	// <gateway:experimental>
+	Timeouts *HTTPRouteTimeouts `json:"timeouts,omitempty"`
+}
+
+// HTTPRouteTimeouts defines timeouts that can be configured for an HTTPRoute.
+// Timeout values are formatted like 1h/1m/1s/1ms, as specified in GEP-2257,
+// and MUST BE >= 1ms or "0s" to disable (no timeout).
+//
+// +kubebuilder:validation:XValidation:message="backendRequest timeout cannot be longer than request timeout",rule="!(has(self.request) && has(self.backendRequest) && duration(self.request) != duration('0s') && duration(self.backendRequest) > duration(self.request))"
+type HTTPRouteTimeouts struct {
+	// Request specifies the duration for processing an HTTP client request after which the
+	// gateway will time out if unable to send a response.
+	//
+	// For example, setting the `rules.timeouts.request` field to the value `10s` in an
+	// `HTTPRoute` will cause a timeout if a client request is taking longer than 10 seconds
+	// to complete.
+	//
+	// This timeout is intended to cover as close to the whole request-response transaction
+	// as possible although an implementation MAY choose to start the timeout after the entire
+	// request stream has been received instead of immediately after the transaction is
+	// initiated by the client.
+	//
+	// When this field is unspecified, request timeout behavior is implementation-dependent.
+	//
+	// Support: Extended
+	//
+	// +optional
+	Request *Duration `json:"request,omitempty"`
+
+	// BackendRequest specifies a timeout for an individual request from the gateway
```

Comment:
Nit: Recommend capitalizing resource names throughout the spec, will not add duplicate comments everywhere though.
```suggestion
	// BackendRequest specifies a timeout for an individual request from the Gateway
```

Reply:
@robscott Are you sure about this? We discussed this before and I argue that if the doc is referring to the gateway instance/deployment, as opposed to the configuration resource, we use lower case. In this case it's not the Gateway configuration resource that is sending the request, it's the gateway instance.

Reply:
Good catch, it feels really funny to see `gateway` instead of `Gateway` but I get your point here. Will go ahead and resolve this one.